cmd/private-prow-configs-mirror/main.go

```diff
@@ -0,0 +1,437 @@
+		return o.isOfficialRepo(orgRepoList[0], orgRepoList[1])
+	}
+	return false
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.ciOpConfigDir, "ci-op-config-path", "", "Path to directory containing ci-operator configurations")
+	fs.StringVar(&o.prowConfig, "prow-config", "", "Prow's configuration file")
+	fs.StringVar(&o.prowPlugins, "prow-plugins", "", "Prow's plugins configuration file")
+
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func (o *options) validate() error {
+	if len(o.ciOpConfigDir) == 0 {
+		return errors.New("--ci-op-config-path is not defined")
+	}
+
+	if len(o.prowConfig) == 0 {
+		return errors.New("--prow-config is not defined")
+	}
+
+	if len(o.prowPlugins) == 0 {
+		return errors.New("--prow-plugins is not defined")
+	}
+	return nil
+}
+
+func loadProwPlugins(pluginsPath string) (*plugins.Configuration, error) {
+	agent := plugins.ConfigAgent{}
+	if err := agent.Load(pluginsPath, false); err != nil {
+		return nil, err
+	}
+	return agent.Config(), nil
+}
+
+func updateProwConfig(configFile string, config prowconfig.ProwConfig) error {
+	data, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("could not marshal Prow configuration: %v", err)
+	}
+
+	return ioutil.WriteFile(configFile, data, 0644)
+}
+
+func updateProwPlugins(pluginsFile string, config *plugins.Configuration) error {
+	data, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("could not marshal Prow configuration: %v", err)
+	}
+
+	return ioutil.WriteFile(pluginsFile, data, 0644)
+}
+
+func privateOrgRepo(repo string) string {
+	return fmt.Sprintf("%s/%s", openshiftPrivOrg, repo)
+}
+
+func getOrgReposWithOfficialImages() (map[string]sets.String, error) {
+	ret := make(map[string]sets.String)
+
+	callback := func(c *api.ReleaseBuildConfiguration, i *config.Info) error {
+		if !promotion.BuildsOfficialImages(c) {
+			return nil
+		}
+
+		if _, ok := ret[i.Org]; !ok {
+			ret[i.Org] = sets.NewString(i.Repo)
+		} else {
+			ret[i.Org].Insert(i.Repo)
+		}
+
+		return nil
+	}
+
+	if err := config.OperateOnCIOperatorConfigDir(config.CiopConfigInRepoPath, callback); err != nil {
+		return ret, fmt.Errorf("error while operating in ci-operator configuration files: %v", err)
+	}
+
+	return ret, nil
+}
+
+func injectPrivateBranchProtection(branchProtection prowconfig.BranchProtection, orgRepos orgReposWithOfficialImages) {
+	privateOrg := prowconfig.Org{
+		Repos: make(map[string]prowconfig.Repo),
+	}
+
+	logrus.Info("Processing...")
+	for orgName, orgValue := range branchProtection.Orgs {
+		for repoName, repoValue := range orgValue.Repos {
+			if orgRepos.isOfficialRepo(orgName, repoName) {
+				logrus.WithField("repo", repoName).Info("Found")
+				privateOrg.Repos[repoName] = repoValue
+			}
+		}
+
+	}
+
+	if len(privateOrg.Repos) > 0 {
+		logrus.WithField("value", privateOrg).Info("Generating openshift-priv org.")
```

Comment:
Given `privateOrg` is a deep object, I think logging it verbatim is not very useful... Personally I think we don't even need this log at all (in all of these methods).